examples/inference.py

```diff
@@ -0,0 +1,161 @@
+from oml.registry.models import get_extractor_by_cfg
+from oml.registry.transforms import get_transforms_by_cfg
+from oml.utils.images.images import imread_cv2, verify_image_readable
+from oml.utils.misc import dictconfig_to_dict
+
+
+def inference(cfg: TCfg) -> None:
+    """
+    This is an entrypoint for the model validation in metric learning setup.
+
+    The config can be specified as a dictionary or with hydra: https://hydra.cc/.
+    For more details look at ``examples/README.md``
+
+    """
+    cfg = dictconfig_to_dict(cfg)
+    pprint(cfg)
+
+    engine_params = parse_engine_params_from_config(cfg)
+    accelerator = engine_params["accelerator"]
+    devices = engine_params["devices"]
+    if accelerator == "gpu":
+        if isinstance(devices, list):
+            device_idx = devices[0]
+        else:
+            # foolprof against selection of multiple devices
+            available_devices = os.getenv("CUDA_VISIBLE_DEVICES", "0")
+            device_idx = int(available_devices.split(",")[0])
+        device = f"cuda:{device_idx}"
+    else:
+        device = accelerator
+
+    images_folder = Path(cfg["images_folder"]) if cfg.get("images_folder") else None
+    dataframe_name = cfg.get("dataframe_name")
+    # Check if either images_folder present or dataframe_name but not both
+    if images_folder is not None and dataframe_name is not None:
+        raise InferenceConfigError("images_folder and dataframe_name cannot be provided both at the same time")
+    if images_folder is None and dataframe_name is None:
+        raise InferenceConfigError("Either images_folder or dataframe_name should be set")
+
+    # Working with pure images if images folder is not None
+    bboxes = None
+    if images_folder is not None:
+        extensions = (".jpeg", ".jpg", ".png")
+
+        im_paths: List[Path] = []
+        for path in sorted(images_folder.rglob("*")):
+            if path.suffix.lower() in extensions and path.is_file():
+                im_paths.append(path)
+
+    # Working with dataframe if it's path is not None
+    if dataframe_name is not None:
+        delimiter = cfg.get("dataframe_separator")
+        # read dataframe and get boxes. Use BaseDataset for inspiration
+        df = pd.read_csv(dataframe_name, delimiter=delimiter)
+        # begin check that columns are correct
+        if PATHS_COLUMN not in df.columns:
+            raise InvalidDataFrameColumnsException(
+                f"Source CSV file {dataframe_name} is missing '{PATHS_COLUMN}' column."
+            )
+
+        if set(df.columns).intersection(set(BBOXES_COLUMNS)):
+            if not all(col in df.columns for col in BBOXES_COLUMNS):
+                raise InvalidDataFrameColumnsException(
+                    f"Boxes are invalid: {df.columns}, please check that all of them exist and have "
+                    f"correct names like this: {BBOXES_COLUMNS}"
+                )
+
+            bboxes = []
```

Comment:
It's set to None above, because that way I don't have to do it in case the source of images is path. But here it's set to list because we then iterate over columns that should have boxes